src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java

```diff
@@ -93,13 +93,9 @@ public void buy(Player player, ShopItem shopItem, int amount) {
         BuyCost buyCost = shopItem.buyCost;
         double vaultCost = calculateCost(amount, shopItem.defaultAmount, buyCost.vault);
         int crystalCost = (int) calculateCost(amount, shopItem.defaultAmount, buyCost.crystals);
+        final Island island = IridiumSkyblockAPI.getInstance().getUser(player).getIsland().get();
```

Comment:
Use IridiumSkyblock#getUserManager Also island can be empty you're getting it while not checked

Reply:
Don't blame me, That was in there before I edited it and I also don't believe a player can open the shop if they don't have an Island

Reply:
u're just idiot!
may they can get kicked island while they're viewing shop ??

Reply:
Okay?? It's literally your code you baboon, I wouldn't have done it the way it is, Your base code not mine you fool